src/commands/selectTemplate.ts

```diff
@@ -0,0 +1,107 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/* eslint-disable no-template-curly-in-string */
+
+import * as vscode from 'vscode';
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ext } from '../extensionVariables';
+import { resolveVariables } from '../utils/resolveVariables';
+
+export type TemplateCommand = 'build' | 'run' | 'runInteractive' | 'attach' | 'logs' | 'composeUp' | 'composeDown';
+
+type CommandTemplate = {
+    template: string,
+    label?: string,
+    match?: string,
+};
+
+// NOTE: the default templates are duplicated in package.json, since VSCode offers no way of looking up extension-level default settings
+// So, when modifying them here, be sure to modify them there as well!
+const defaults: { [key in TemplateCommand]: CommandTemplate } = {
+    'build': { label: 'Docker Build', template: 'docker build --rm -f "${dockerfile}" -t ${tag} "${context}"' },
+    'run': { label: 'Docker Run', template: 'docker run --rm -d ${exposedPorts} ${tag}' },
+    'runInteractive': { label: 'Docker Run (Interactive)', template: 'docker run --rm -it ${exposedPorts} ${tag}' },
+    'attach': { label: 'Docker Attach', template: 'docker exec -it ${containerId} ${shellCommand}' },
+    'logs': { label: 'Docker Logs', template: 'docker logs -f ${containerId}' },
+    'composeUp': { label: 'Compose Up', template: 'docker-compose up ${configurationFile} ${detached} ${build}' },
+    'composeDown': { label: 'Compose Down', template: 'docker-compose down ${configurationFile}' },
+};
+
+export async function selectTemplate(context: IActionContext, command: TemplateCommand, matchContext?: string, folder?: vscode.WorkspaceFolder, additionalVariables?: { [key: string]: string }): Promise<string> {

```

Comment:
Should `matchContext` be an array of strings?  While unlikely, there's a chance that a matching template could match content that overlaps individual context values.  It seems like it'd be better to test each piece of context individually.